content/id/docs/tasks/administer-cluster/dns-custom-nameservers.md

```diff
@@ -0,0 +1,262 @@
+* [loop](https://coredns.io/plugins/loop/): Mendeteksi _forwarding loop_ sederhana dan menghentikan proses CoreDNS jika _loop_ ditemukan.
+* [reload](https://coredns.io/plugins/reload): Mengizinkan _reload_ automatis Corefile yang telah diubah. Setelah kamu mengubah konfigurasi ConfigMap, beri waktu sekitar dua menit agar perubahan yang kamu lakukan berlaku.
+* [loadbalance](https://coredns.io/plugins/loadbalance): Ini adalah _load balancer_ DNS secara _round-robin_ yang mengacak urutan _record_ A, AAAA, dan MX dalam setiap responnya.
+
+Kamu dapat memodifikasi perilaku CoreDNS bawaan dengan memodifikasi ConfigMap.
+
+### Konfigurasi _Stub-domain_ dan _Nameserver Upstream_ dengan menggunakan CoreDNS
+
+CoreDNS memiliki kemampuan untuk mengonfigurasi _stubdomain_ dan _nameserver upstream_ dengan menggunakan [_plugin_ forward](https://coredns.io/plugins/forward/).
+
+#### Contoh
+
+Jika operator klaster memiliki sebuah server domain [Consul](https://www.consul.io/) yang terletak di 10.150.0.1, dan semua nama Consul memiliki akhiran .consul.local. Untuk mengonfigurasinya di CoreDNS, administrator klaster membuat bait (_stanza_) berikut dalam ConfigMap CoreDNS.
+
+```
+consul.local:53 {
+        errors
+        cache 30
+        forward . 10.150.0.1
+    }
+```
+
+Untuk memaksa secara eksplisit semua pencarian DNS _non-cluster_ melalui _nameserver_ khusus pada 172.16.0.1, arahkan `forward` ke _nameserver_ bukan ke `/etc/resolv.conf`
+
+```
+forward .  172.16.0.1
+```
+
+ConfigMap terakhir bersama dengan konfigurasi `Corefile` bawaan terlihat seperti berikut:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: coredns
+  namespace: kube-system
+data:
+  Corefile: |
+    .:53 {
+        errors
+        health
+        kubernetes cluster.local in-addr.arpa ip6.arpa {
+           pods insecure
+           fallthrough in-addr.arpa ip6.arpa
+        }
+        prometheus :9153
+        forward . 172.16.0.1
+        cache 30
+        loop
+        reload
+        loadbalance
+    }
+    consul.local:53 {
+        errors
+        cache 30
+        forward . 10.150.0.1
+    }
+```
+
+Perangkat `kubeadm` mendukung terjemahan automatis dari ConfigMap kube-dns
+ke ConfigMap CoreDNS yang setara.
+
+{{< note >}}
+Sementara ini kube-dns dapat menerima FQDN untuk _stubdomain_ dan _nameserver_ (mis: ns.foo.com), namun CoreDNS belum mendukung fitur ini.
+Selama penerjemahan, semua _nameserver_ FQDN akan dihilangkan dari konfigurasi CoreDNS.
+{{< /note >}}
+
+## Konfigurasi CoreDNS yang setara dengan kube-dns
+
+CoreDNS mendukung fitur kube-dns dan banyak lagi lainnya.
+ConfigMap dibuat agar kube-dns mendukung `StubDomains` dan `upstreamNameservers` untuk diterjemahkan ke _plugin_ `forward` dalam CoreDNS.
+Begitu pula dengan _plugin_ `Federations` dalam kube-dns melakukan translasi untuk _plugin_ `federation` dalam CoreDNS.
+
+### Contoh
+
+Contoh ConfigMap ini untuk kube-dns menentukan federasi, _stubdomain_ dan server _upstreamnameserver_:
```

Comment:
nit: should we add space for readability?
```suggestion
Contoh ConfigMap ini untuk kube-dns menentukan federasi, _stub domain_ dan server _upstream nameserver_:
```

Reply:
Ok!